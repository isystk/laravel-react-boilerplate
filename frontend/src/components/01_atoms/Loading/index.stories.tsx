import { Meta, Story } from '@storybook/react'
import React from 'react'
import Loading, { LoadingProps } from './index'

export default {
  title: '01_atoms/Loading',
  component: Loading,
} as Meta

const Template: Story = () => {
  const props: LoadingProps = {
    loading: true,
  }
  return <Loading {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
