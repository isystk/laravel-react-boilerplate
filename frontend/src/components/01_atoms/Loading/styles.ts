import { css, keyframes } from '@emotion/css'

export const loading = css``
