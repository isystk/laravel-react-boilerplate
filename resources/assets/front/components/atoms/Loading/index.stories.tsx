import Loading from './index';
import type { Meta } from '@storybook/react';
import { JSX, useEffect } from 'react';
import { AppProvider } from '@/states/AppContext';
import useAppRoot from '@/states/useAppRoot';

export default {
  title: 'Components/Atoms/Loading',
  component: Loading,
  tags: ['autodocs'],
  decorators: [
    Story => (
      <AppProvider>
        <Story />
      </AppProvider>
    ),
  ],
} as Meta<typeof Loading>;

export const Default: { render: () => null | JSX.Element } = {
  render: () => {
    const { state, service } = useAppRoot();
    useEffect(() => {
      if (state) {
        service.showLoading();
      }
    }, [state, service]);
    if (!state) return null;
    return <Loading />;
  },
};
