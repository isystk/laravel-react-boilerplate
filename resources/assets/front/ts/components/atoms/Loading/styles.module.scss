.overlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(255, 255, 255, 0.8);
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    z-index: 9999;
}

.spinner {
    width: 50px;
    height: 50px;
    border: 5px solid #ccc;
    border-top: 5px solid #333;
    border-radius: 50%;
    animation: spin 1s linear infinite;
}

.message {
    margin-top: 16px;
    font-size: 16px;
    color: #333;
}

@keyframes spin {
    to {
        transform: rotate(360deg);
    }
}
