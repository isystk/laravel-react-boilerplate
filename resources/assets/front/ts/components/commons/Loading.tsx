import React, { FC } from "react";
import MainService from "@/services/main";
import { Spinner } from "reactstrap";
import Portal from "@/components/commons/Portal";

type Props = {
    appRoot: MainService;
};

const Loading: FC<Props> = ({ appRoot }) => {
    const { isShowLoading } = appRoot;
    return (
        <Portal>
            {isShowLoading && (
                <div id="site_loader_overlay">
                    <div className="site_loader_spinner">
                        <Spinner>Loading...</Spinner>
                    </div>
                </div>
            )}
        </Portal>
    );
};

export default Loading;
