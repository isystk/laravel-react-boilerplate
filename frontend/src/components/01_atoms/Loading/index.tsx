import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'

/** LoadingProps Props */
export type LoadingProps = WithChildren & {
  loading: boolean
}
/** Presenter Props */
export type PresenterProps = LoadingProps

/** Presenter Component */
const LoadingPresenter: FC<PresenterProps> = ({ loading }) => (
  <>
    <div
      className={`${styles.loading} fixed top-0 left-0 w-full h-full z-30 ${
        loading ? '' : 'hidden'
      }`}
    >
      <div className="absolute bottom-0 right-0 p-8 flex justify-center">
        <div className="animate-spin h-10 w-10 border-4 border-gray-500 rounded-full border-t-transparent"></div>
      </div>
    </div>
  </>
)

/** Container Component */
const LoadingContainer: React.FC<
  ContainerProps<LoadingProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  return presenter({
    children,
    ...props,
  })
}

export default connect<LoadingProps, PresenterProps>(
  'Loading',
  LoadingPresenter,
  LoadingContainer
)
