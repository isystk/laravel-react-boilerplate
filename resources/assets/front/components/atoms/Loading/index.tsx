import styles from './styles.module.scss';
import useAppRoot from '@/states/useAppRoot';
import Portal from '@/components/interactions/Portal';

const Loading = () => {
  const [state] = useAppRoot();
  if (!state) return null;

  const { isLoading } = state;

  return (
    <Portal>
      <>
        {isLoading && (
          <div className={styles.overlay}>
            <div className={styles.spinner}></div>
            <p id="loading" className={styles.message}>
              Loading...
            </p>
          </div>
        )}
      </>
    </Portal>
  );
};

export default Loading;
