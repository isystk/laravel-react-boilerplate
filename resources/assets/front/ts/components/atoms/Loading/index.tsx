import styles from './styles.module.scss';
import useAppRoot from '@/stores/useAppRoot';
import Portal from '@/components/interactions/Portal';

const Loading = () => {
  const appRoot = useAppRoot();
  if (!appRoot) return null;

  const { isShowLoading } = appRoot;

  return (
    <Portal>
      <>
        {isShowLoading && (
          <div className={styles.overlay}>
            <div className={styles.spinner}></div>
            <p className={styles.message}>Loading...</p>
          </div>
        )}
      </>
    </Portal>
  );
};

export default Loading;
