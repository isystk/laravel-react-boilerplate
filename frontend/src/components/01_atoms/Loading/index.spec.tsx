import React from 'react'
import renderer from 'react-test-renderer'
import Loading, { LoadingProps } from './index'

describe('Loading', () => {
  it('Match Snapshot', () => {
    const props: LoadingProps = {
      loading: true,
    }
    const component = renderer.create(<Loading {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
