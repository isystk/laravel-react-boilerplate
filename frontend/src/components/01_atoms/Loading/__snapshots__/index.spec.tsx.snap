// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Match Snapshot 1`] = `
<div
  className="css-0 fixed top-0 left-0 w-full h-full z-30 "
>
  <div
    className="absolute bottom-0 right-0 p-8 flex justify-center"
  >
    <div
      className="animate-spin h-10 w-10 border-4 border-gray-500 rounded-full border-t-transparent"
    />
  </div>
</div>
`;
